tasks/tasks_test.go

```diff
@@ -0,0 +1,67 @@
+package tasks
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTaskErrorLeadsToErrorStatus(t *testing.T) {
+	assert := assert.New(t)
+	errorTask := func() error {
+		return errors.New("This is the error text.")
+	}
+	ex := NewTaskExectuer()
+	id := ex.QueueTask(errorTask)
+	status, err := ex.GetTaskStatus(id)
+	for status == INPROGRESS {
+		status, err = ex.GetTaskStatus(id)
+	}
+	assert.Nil(err)
+	assert.Equal(FAILURE, status)
```

Comment:
That's pretty weird. You should be running `go test ./...` from the top-level directory. Do you still get the error?